src/main/java/uk/ac/ebi/atlas/experimentpage/ExpressionAtlasContentService.java

```diff
@@ -175,50 +175,53 @@ public ExpressionAtlasContentService(
-
-        if (experiment.getType().isProteomicsBaseline()) {
-            return proteomicsBaselineExperimentExternallyAvailableContentService.list(
-                    (BaselineExperiment) experiment, contentType);
-        } else if (experiment.getType().isRnaSeqBaseline()) {
-            List list = rnaSeqBaselineExperimentExternallyAvailableContentService.list(
-                    (BaselineExperiment) experiment, contentType);
-            List list2 =  externalResourceType.equals("geo") ?
-                    rnaSeqBaselineExperimentExternallyAvailableContentServiceGeo.list(
-                    (BaselineExperiment) experiment, contentType) :
-                    externalResourceType.equals("ega") ?
-                    rnaSeqBaselineExperimentExternallyAvailableContentServiceEga.list(
-                            (BaselineExperiment) experiment, contentType) :
-                    rnaSeqBaselineExperimentExternallyAvailableContentServiceEna.list(
-                            (BaselineExperiment) experiment, contentType);
-            list.addAll(list2);
-            return list;
-        } else if (experiment.getType().isRnaSeqDifferential()) {
-            List list = rnaSeqDifferentialExperimentExternallyAvailableContentService.list(
-                    (DifferentialExperiment) experiment, contentType);
-            List list2 =  externalResourceType.equals("geo") ?
-                    rnaSeqDifferentialExperimentExternallyAvailableContentServiceGeo.list(
-                            (DifferentialExperiment) experiment, contentType) :
-                    externalResourceType.equals("ega") ?
-                    rnaSeqDifferentialExperimentExternallyAvailableContentServiceEga.list(
-                            (DifferentialExperiment) experiment, contentType) :
-                    rnaSeqDifferentialExperimentExternallyAvailableContentServiceEna.list(
-                            (DifferentialExperiment) experiment, contentType);
-            list.addAll(list2);
-            return list;
-        } else {
-            List list = microarrayExperimentExternallyAvailableContentService.list(
-                    (MicroarrayExperiment) experiment, contentType);
-            List list2 =  externalResourceType.equals("geo") ?
-                    microarrayExperimentExternallyAvailableContentServiceGeo.list(
-                            (MicroarrayExperiment) experiment, contentType) :
-                    externalResourceType.equals("ega") ?
-                    microarrayExperimentExternallyAvailableContentServiceEga.list(
-                            (MicroarrayExperiment) experiment, contentType) :
-                    microarrayExperimentExternallyAvailableContentServiceEna.list(
-                            (MicroarrayExperiment) experiment, contentType);
-            list.addAll(list2);
-            return list;
+        List arrayExpressLinks;
+        List otherExternalResourceLinks;
```

Comment:
Don’t do this! At this point both `arrayExpressLinks` and `otherExternalResourceLinks` are `null`, and that’s asking for trouble.

A safe alternative is an `ImmutableList.Builder` which you populate in each case below.